TACHO_DRIVER.md

```diff
@@ -0,0 +1,199 @@
+attributes while the motor is running.
+
+If you change the allowed `duty_cycle_sp` or `speed_sp` value while the
+motor is in `run_direct` mode, the ramp time setpoints govern the operation.
+
+No matter the sign of the `duty_cycle_sp` or `speed_sp`, if the new
+setpoint is towards a lower absolute value, then the `ramp_down_sp`
+determines how the motor slows down. If the new setpoint is towards a
+higher absolute value, then the `ramp_up_sp` determines how the motor
+speeds up.
+
+An example follows:
+
+- `ramp_up_sp` is set to 4000
+- `ramp_down_sp` is set to 2000
+- `speed_regulation` is set to `off`
+- `duty_cycle_sp` is set to 75%
+
+If we write `run-direct` to the motor `command` attribute, the motor will
+take 3000 msec to ramp up to 75% duty cycle. What happens when we change the
+`duty_cycle_sp` to other values while the motor is running?
+
+| New `duty_cycle_sp` | Operation
+| ------------------- | ----------------------
+|   100               | Motor ramps up to 100% in 1000 msec
+|   50                | Motor ramps down to 50% in 1000 msec
+|   -50               | Motor ramps down to 0% in 1000 msec and up to -50 in 2000 msec
+
+## Run Modes
+
+The `run` attribute determines function that the motor is executing. Within a `run`
+mode, some attributes have an effect, while others do not. These details are
+described below.
+
+The run modes can be classified into two types. The `run-direct` is suited to
+applications where the programmer wants direct control of the motors. The other
+three modes:
+
+- `run_timed`
+- `run_to_abs_pos`
+- `run_to_rel_pos`
+
+They can be through of as convenience commands where the driver handles the motor
+control independently of the application. This is well suited to applications where you
+want the driver to handle the details, like machines where the motors must alternate
+between precise positions or run for fixed timing periods.
+
+### `run-direct` Mode
+
+In its most basic operating mode, the `tacho-motor` accepts the `run-direct`
+command. After accepting a `run-direct` command, the motor will only accept
+a `stop` command - the other commands represent "helpers" that automatically
+do what `run-direct` is capable of without user intervention.
+
+NOTE: In this mode the `stop` command stops the motor immediately! If you
+want the motor to use the ramping feature, change the desired setpoint to 0.
+
+Currently, only changes to the `duty_cycle_sp` take effect
+immediately and I'm proposing that this should depend on the `speed_regulation`
+setting. I may be convinced to also allow changes to the ramp setpoints.
+
+You cannot change the `speed_regulation` value without stopping the
+motor first. (Not sure on this - I can be convinced to change this)
+
+Finally, if the ramping target setpoint has been set to 0, the motor is
+stopped in the corresponding `stop_mode`.
+
+### `run-forever` Mode
+
+Deprecated? This does not provide any new features that are not already present
+in `run-direct`
```

Comment:
PiStorms only has speed regulated mode. You cannot directly control the duty cycle output. (unless you convince midsensors to implement it in the firmware and get everyone that already has one to upgrade their firmware).

As I explained [here](https://github.com/ev3dev/ev3dev/issues/453#issuecomment-168771926), I think `run-direct` should be a special case for advanced users that want to write their own PID and have super-fast response to changes in `duty_cycle_sp`. Most normal users would never use `run-direct`.

I already explained above, but `run-direct` only looks at `duty_cycle_sp`. All other attributes are ignored. `run-forever` respects all attributes. Also `run-direct` is not technically possible on all platforms/motor controllers whereas `run-forever` is.